crates/autopilot/src/domain/competition/winner_selection/combinatorial.rs

```diff
@@ -350,3 +350,728 @@ type ScoreByDirection = HashMap<DirectedTokenPair, Score>;
+                    },
+                    "score": score(100),
+                }
+            },
+            "expected_fair_solutions": ["Solution 1", "Solution 2",  "Solution 3"],
+            "expected_winners": ["Solution 1"],
+            "expected_reference_scores": {
+                "Solver 1": "380",
+                "Solver 2": "300",
+                "Solver 3": "300",
+            },
+        });
+        TestCase::from_json(case).validate();
+    }
+
+    #[test]
+    fn staging_mainnet_auction_12825008() {
+        // https://solver-instances.s3.eu-central-1.amazonaws.com/staging/mainnet/autopilot/12825008.json
+        // The example is an auction with one order and two competing bids for it, one
+        // having a better score than the other
+
+        let case = json!({
+            "tokens": [
+                // corresponding to 0x67466be17df832165f8c80a5a120ccc652bd7e69
+                ["Token A", create_address(0)],
+                // corresponding to 0xdac17f958d2ee523a2206206994597c13d831ec7
+                ["Token B", create_address(1)],
+            ],
+            "auction": {
+                "orders": {
+                    "Order 1": ["Token A", "32375066190000000000000000", "Token B", "2161512119"],
+                },
+                "prices": {
+                    "Token A": "32429355240",
+                    "Token B": "480793239987749750742974464",
+                }
+            },
+            "solutions": {
+                // solution 1 (baseline, the winner)
+                "Solution 1": {
+                    "solver": "Solver 1",
+                    "trades": {
+                        "Order 1": ["32375066190000000000000000", "2206881314"],
+                    },
+                    "score": "21813202259686016",
+                },
+                // solution 2 (zeroex)
+                "Solution 2": {
+                    "solver": "Solver 2",
+                    "trades": {
+                        "Order 1": ["32375066190000000000000000", "2205267875"],
+                    },
+                    "score": "21037471695353421",
+                },
+            },
+            "expected_fair_solutions": ["Solution 1", "Solution 2"],
+            "expected_winners": ["Solution 1"],
+            "expected_reference_scores": {
+                "Solver 1": "21037471695353421",
+            },
+        });
+        TestCase::from_json(case).validate();
+    }
+
+    #[serde_as]
+    #[derive(Deserialize, Debug)]
+    pub struct TestCase {
+        pub tokens: Vec<(String, H160)>,
+        pub auction: TestAuction,
+        pub solutions: HashMap<String, TestSolution>,
+        pub expected_fair_solutions: Vec<String>,
+        pub expected_winners: Vec<String>,
+        // workaoround needed as JSON have a limitation on number size, so we accept numbers as
+        // String
+        #[serde_as(as = "HashMap<_, DisplayFromStr>")]
+        pub expected_reference_scores: HashMap<String, u128>,
```

Comment:
nit: technically this should be a `U256` which can also be deserialized from a string.